core/ArrayLib.php

```diff
@@ -164,6 +164,23 @@ public static function in_array_recursive($needle, $haystack, $strict = false) {
 	}
 
 	/**
+	 * Similar to array_map, but recurses when arrays are encountered.
+	 *
+	 * Actually only one array argument is supported.
+	 *
+	 * @param $f callback to apply
+	 * @param $array array
+	 * @return array
+	 */
+	public static function array_map_recursive($f, $array) {
+		$applyOrRecurse = function($v) use($f) {
+			return is_array($v) ? ArrayLib::array_map_recursive($f, $v) : $f($v);
```

Comment:
I would suggest `call_user_func($f)` over `$f`, since this extends the ability of this method.

For instance, I might want to pass in `array($this, 'mymapmethod)` to the first parameter.